recipe-server/client/control/components/Fields.js

```diff
@@ -39,10 +40,65 @@ export function buildControlField({
 buildControlField.propTypes = {
   input: pt.object.isRequired,
   meta: pt.shape({
-    error: pt.string,
+    error: pt.oneOfType([pt.string, pt.array]),
   }).isRequired,
   label: pt.string.isRequired,
   className: pt.string,
   InputComponent: pt.oneOfType([pt.func, pt.string]),
+  hideErrors: pt.bool,
   children: pt.node,
 };
+
+export function IntegerControlField(props) {
+  return (
+    <ControlField
+      component="input"
+      type="number"
+      step="1"
+      parse={value => {
+        try {
```

Comment:
try/catches are slow in JS - could we use underscore's `isNumber` instead? ex:

`...parse={ value => _.isNumber(value) ? parseInt(value, 10) : 0 }...`